packages/firestore/test/unit/local/remote_document_cache.test.ts

```diff
@@ -42,25 +57,55 @@ describe('IndexedDbRemoteDocumentCache', () => {
     return;
   }
 
-  genericRemoteDocumentCacheTests(() =>
-    persistenceHelpers.testIndexedDbPersistence(/* synchronizeTabs= */ true)
-  );
+  beforeEach(async () => {
+    persistence = await persistenceHelpers.testIndexedDbPersistence({
+      synchronizeTabs: true
+    });
+  });
+
+  afterEach(() => persistence.shutdown(/* deleteData= */ true));
+
+  it('can prune change log', async () => {
+    // Add two change batches and remove the first one.
+    await persistence.runTransaction(
+      'removeDocumentChangesThroughChangeId',
+      true,
+      txn => {
+        const cache = persistence.getRemoteDocumentCache() as IndexedDbRemoteDocumentCache;
+        return cache
+          .addEntries(txn, [doc('a/1', 1, DOC_DATA), doc('b/1', 2, DOC_DATA)])
+          .next(() => cache.getNewDocumentChanges(txn))
+          .next(() => cache.addEntries(txn, [doc('c/1', 3, DOC_DATA)]))
+          .next(() => cache.removeDocumentChangesThroughChangeId(txn, 1));
+      }
+    );
+    await persistence.shutdown(/* deleteData= */ false);
+
+    // Now make sure that we can only read back the second batch.
+    persistence = await persistenceHelpers.testIndexedDbPersistence({
+      synchronizeTabs: true,
+      dontPurgeData: true
+    });
+    const changedDocs = await persistence.runTransaction(
+      'getNewDocumentChanges',
+      false,
+      txn => {
+        const cache = persistence.getRemoteDocumentCache();
+        return cache.getNewDocumentChanges(txn);
```

Comment:
So looking at this test it occurs to me to ask:  If we only GC these every 30 minutes, does that mean when you open a new tab it's going to have to replay 30 minutes of changes on start?

Reply:
This is probably not the cleanest test, but it doesn't call start() on the RemoteDocumentCache. start() forwards the internal pointer to the current location in the DB, and that means that older changes won't get replayed.

Short of fixing the test, I added a comment to explain this behavior.